server/aws-lsp-codewhisperer/src/experimental/amazonQCompletionServer.ts

```diff
@@ -0,0 +1,333 @@
+                            leftFileContent: requestContext.fileContext.leftFileContent
+                                .slice(-CONTEXT_CHARACTERS_LIMIT)
+                                .replaceAll('\r\n', '\n'),
+                            rightFileContent: requestContext.fileContext.rightFileContent
+                                .slice(0, CONTEXT_CHARACTERS_LIMIT)
+                                .replaceAll('\r\n', '\n'),
+                        },
+                    })
+
+                    const filteredSuggestions = suggestionResponse.suggestions
+                        // Empty suggestion filter
+                        .filter(suggestion => {
+                            if (suggestion.content === '') {
+                                return false
+                            }
+
+                            return true
+                        })
+                        // References setting filter
+                        .filter(suggestion => {
+                            if (includeSuggestionsWithCodeReferences) {
+                                return true
+                            }
+
+                            if (suggestion.references == null || suggestion.references.length === 0) {
+                                return true
+                            }
+
+                            // Filter out suggestions that have references when includeSuggestionsWithCodeReferences setting is true
+                            return false
+                        })
+
+                    // If after all server-side filtering no suggestions can be displayed, close session and return empty results
+                    if (filteredSuggestions.length === 0) {
+                        logging.log('All suggestions were filtered out.')
+
+                        return EMPTY_RESULT
+                    }
+
+                    // Get content from the left of the cursor to generate full line
+                    const left = textDocument.getText({
+                        start: { line: params.position.line, character: 0 },
+                        end: params.position,
+                    })
+
+                    const completionItems = mergeSuggestionsWithRightContext(
+                        fileContext.rightFileContent,
+                        filteredSuggestions
+                    ).map(item => {
+                        const label = (left + item.insertText).trimStart()
+                        return {
+                            label: label,
+                            // insertText: item.insertText as string,
+                            textEdit: {
+                                range: {
+                                    start: params.position,
+                                    end: params.position,
+                                },
+                                newText: item.insertText as string,
+                            },
+                            kind: CompletionItemKind.Text,
```

Comment:
Discussion with Neovim plugin authors leads me to believe that "snippet" is most appropriate for multiline completions. 


```suggestion
                            kind: CompletionItemKind.Snippet,
```

Reply:
Actually based on feedback from blink.nvim users, `usekind: CompletionItemKind.Snippet` seems to work better for them, while `insertTextFormat: InsertTextFormat.Snippet` risks activating snippet "placeholder" handling (snippet mode).

Reply:
I'm not sure if using Snippet type would be correct, given it expects that returned text has richer syntax, defined in LSP https://microsoft.github.io/language-server-protocol/specifications/lsp/3.17/specification/#snippet_syntax. 

I don't have clear overview of how different IDE/LSP Client handle Snippets, I suspect it might cause issues in some clients.

Reply:
Further feedback from Neovim users has indicated that `Text` or `Reference` work better, because completion engines often filter-out `Snippet` (and sometimes `Text`, too).